spec/lib/jobs/rest-job-spec.js

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016, EMC, Inc
+/* jshint node: true */
+
+'use strict';
+
+var nock = require('nock');
+var uuid = require('node-uuid');
+
+describe("REST-job", function(){
+    var RestJob;
+    var restJob;
+    var testUrl = 'https://test.address.com:12345',
+        taskId = uuid.v4(),
+        options = {
+            url: testUrl + '/full/put',
+            method: 'PUT',
+            credential: {username:"foo", password:"bar"},
+            headers: {
+                "content-type": "application/json",
+                "some-token": "whatever-ssl-token"
+                },
+            data: "nobody cares"
+        },
+        context = {};
+
+    before(function(){
+        helper.setupInjector([
+            helper.require('/lib/jobs/rest-job.js'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/utils/job-utils/http-tool.js')
+        ]);
+        RestJob = helper.injector.get('Job.Rest');
+    });
+
+  
+    // With all parameters provided, REST should be good 
+    it('Should do REST with all parameters', function(done){
+        nock(testUrl)
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('some-token', 'whatever-ssl-token')
+        .put('/full/put')
+        .basicAuth({user: 'foo', pass: 'bar'})
+        .reply(201, 'You are good');
+
+        context = {};
+        restJob = new RestJob(options, context, taskId);
+
+        restJob.run().then(function(){
```

Comment:
The promise must be returned otherwise any assertion will not be captured by mocha.